clang/test/CodeGen/fp-atomic-ops.c

```diff
@@ -42,3 +45,22 @@ void test(float *f, float ff, double *d, double dd) {
   __atomic_fetch_sub(d, dd, memory_order_relaxed);
 #endif
 }
+
+typedef float float2 __attribute__((ext_vector_type(2)));
+typedef double double2 __attribute__((ext_vector_type(2)));
+
+void test_vector(float2 *f, float2 ff, double2 *d, double2 dd) {
+  // FLOAT: atomicrmw fadd ptr {{.*}} <2 x float> {{.*}} monotonic
+  __atomic_fetch_add(f, ff, memory_order_relaxed);
+
+  // FLOAT: atomicrmw fsub ptr {{.*}} <2 x float> {{.*}} monotonic
+  __atomic_fetch_sub(f, ff, memory_order_relaxed);
+
+#ifdef DOUBLE
```

Comment:
Don't know why this macro check is here, but the unguarded typedef probably defeats the point. Can we just not have it